craft_providers/lxd/launcher.py

```diff
@@ -136,13 +137,58 @@ def _formulate_base_instance_name(
     return "-".join(["base-instance", compatibility_tag, image_remote, image_name])
 
 
-# TODO: Implement this stubbed function (CRAFT-1341)
-# pylint: disable-next=unused-argument
-def _is_valid(instance: LXDInstance) -> bool:
-    """Check if a instance is valid (i.e. the base instance is not expired).
+def _is_valid(*, instance_name: str, project: str, remote: str, lxc: LXC) -> bool:
+    """Check if a instance is valid.
 
-    :returns: True if the instance is valid.
+    Instances are valid for 3 months (90 days). After 3 months, they are considered
+    expired and invalid.
+
+    If errors occur during the validity check, the instance is assumed to be invalid.
+
+    :param instance: Name of instance to check the validity of.
+    :param project: LXD project name to create.
+    :param remote: LXD remote to create project on.
+    :param lxc: LXC client.
+
+    :returns: True if the instance is valid. False otherwise.
     """
+    logger.debug("Checking validity of instance %r.", instance_name)
+
+    # capture instance info
+    try:
+        info = lxc.info(instance_name=instance_name, project=project, remote=remote)
+    except LXDError as raised:
+        # if the base instance info can't be retrieved, consider it invalid
+        logger.warning("Could not get instance info with error: %s", raised)
+        return False
+
+    creation_date_raw = info.get("Created")
+
+    # if the base instance does not have a creation date, consider it invalid
+    if not creation_date_raw:
+        logger.warning("Instance does not have a 'Created' date.")
+        return False
+
+    # parse datetime
+    try:
+        creation_date = datetime.strptime(creation_date_raw, "%Y/%m/%d %H:%M %Z")
+    except ValueError as raised:
+        # if the date can't be parsed, consider it invalid
+        logger.warning(
+            "Could not parse instance's 'Created' date with error: %s", raised
```

Comment:
Please also log with `%r` the `creation_date_raw`, just try to be as more explicit/precise as possible if a parse error happens...

Reply:
Good idea